articles/automation-testing-in-selenium/index.md

```diff
@@ -0,0 +1,99 @@
+* **Ease of use:** This software can be installed and used easily. 
+* **Low Cost:** This software is open-source, which reduces the cost of licensing and testing. This also provides an open community for providing customer support. 
+* **Flexibility:** Selenium software supports different browsers, programming languages, and browsers. This offers flexibility to web developers. This is unlike other testing software like QTP and RFT, which support fewer number of browsers.
+* **Integration:** It is possible to integrate this software with other applications such as TestNG, Docker, Maven, and Junit. These applications add value to automation testing, especially test management.
+* **Hardware consumption:** The execution of scripts requires low hardware consumption. 
+
+### Components of Selenium and How They Work
+Selenium software has four main components: Integrated Development Environment (IDE), remote control (RC), web driver, and grid. These components perform specific roles in automation testing.
+
+#### Selenium Integrated Development Environment (IDE)
+This can also be termed as a Selenium Recorder. It is utilized in the editing and recording of functional tests. It is also used for replaying and debugging tests. IDE has a playback feature that enables users to playback the recorded tests. In Mozilla Firefox, it is implemented as an add-on. In google chrome, it is made available as a plugin. 
+
+The interactions recorded by Selenium IDE are exported in the form of a reusable script. Users can export test cases to other Selenium components such as Web Driver or Remote Control.    
+
+Selenium IDE can be used even by people who have little conversance with programming languages. Some of the commands in this component include open, ClickAndWait, and assert.  
+
+The following diagram shows a screenshot of an IDE plugin in Mozilla Firefox.
+
+![Selenium IDE](/engineering-education/automation-testing-in-selenium/selenium-ide.png)
+
+[Image Source: Edureka](https://d1jnx9ba8s6j9r.cloudfront.net/blog/wp-content/uploads/2017/04/selenium-ide.png)
+
+When a web browser is opened by a user, the Selenium Recorder records the test cases (automatically). The commands inserted by the IDE depend on the actions of the user. The test records can be configured or turned off, depending on the user's need. The recorded test cases are then edited by inserting editing commands. These records can then be saved by inserting the save command. 
+
+#### Selenium Remote Control (RC)
+This can also be termed as Selenium 1. It is a server that utilizes HTTP protocol to accept the command of browsers. It supports the use of various programming languages such as Python, C#, Java, and Perl. Selenium 1 consists of two main components: the RC server and the RC client. The former uses HTTP requests to communicate. RC client acts as a library for the programming codes.

```

Comment:
Could you add a screenshot to give an overview about Selenium Remote Control?

Reply:
Done.